src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -95,7 +95,13 @@ def run_python_test(transitive_hydrated_target, pytest, python_setup, source_roo
   ]
   requirements_pex_request = ExecuteProcessRequest(
     argv=tuple(requirements_pex_argv),
-    env={'PATH': os.pathsep.join(python_setup.interpreter_search_paths)},
+    env={
+      'PATH': os.pathsep.join(python_setup.interpreter_search_paths),
+      # NB: CPPFLAGS and LDFLAGS are necessary on certain systems (i.e. macOS) to get cryptography
+      # properly resolved.
+      'CPPFLAGS': os.environ.get('CPPFLAGS', ''),
+      'LDFLAGS': os.environ.get('LDFLAGS', ''),
```

Comment:
See Daniel's comment here: https://github.com/pantsbuild/pants/commit/42079bf1ffa3bf7be91c616df1ea95c32d3dd0d9#r33534265

I'm also not super comfortable with this change, but I don't know how else to get Pants working on macOS.

Reply:
@illicitonion : Does the remote execution API specify a way to essentially opt-in to "env" leaks for this use case? Or, if we're assuming docker containers in RBE, is the idea that the client would know the destination host intimately, and set these correctly for that host?

The general idea of a "configurable execution environment" is something that I think might apply to both of these cases. I'm not sure yet what exact API we should be using to specify these kinds of variables, but it seems like we will have two cases:
1. we're executing locally, and so we want to use values that we've inferred based on localhost's values
2. we're executing remotely on a precise docker image, and want to override all of the values for that case

... although it is pretty clearly rough, I think that we might consider breaking out a `Subsystem` to centralize these, such that we always set them, but we switch off which set we're using based on the destination of the process execution.

Reply:
The remote execution API doesn't support any leaking. The idea is that the client should know its execution platform enough to set whatever values it needs. Generally I agree with Stu's comment ^^

Reply:
> "configurable execution environment"

The native backend has examples of exactly this in `native_toolchain.py`, conveniently already packaged as v2 rules, including objects which set precisely `CPPFLAGS` and `LDFLAGS`.